scripts/bootstrap-linux.sh

```diff
@@ -0,0 +1,161 @@
+
+run_command() {
+    if [ "$option_dry_run" = "yes" ]; then
+        command_prefix="(dry-run) "
+    fi
+
+    echo "$command_prefix=> $@"
+
+    if [ "$option_dry_run" = "no" ]; then
+        $@
+    fi
+}
+
+clone_b2g() {
+    echo ""
+    echo "Clone selected"
+    echo ""
+    if [ -d "$PWD/B2G" ]; then
+        echo "Found existing B2G: $PWD/B2G"
+        return 0
+    fi
+    run_command git clone git://github.com/mozilla-b2g/B2G.git
+}
+
+auto_install() {
+	echo ""
+	echo "Auto install selected"
+	echo ""
+	# Validate the host OS and perform platform specific validation.
+	os=$(grep -R 'NAME=' /etc/os-release | head -1 | cut -d '"' -f 2)
+	echo "$os operating system detected"
+
+	# Change the variable contents and functions to match the OS.
+	if [ "$os" = "Ubuntu" ] || [ "$os" = "Debian" ]; then
+		validate_list="autoconf2.13 bison bzip2 ccache curl flex g++ g++-4.6 g++-4.6-multilib gawk gcc gcc-4.6 git g++-multilib ia32-libs lib32ncurses5-dev lib32z1-dev libgl1-mesa-dev libx11-dev libxml2-utils make zip zlib1g:amd64 zlib1g-dev:amd64 zlib1g-dev:i386 zlib1g:i386"
+		validate_command(){
+			package="$1"
+			echo $(	apt-cache policy "$package" | grep 'Installed:' | grep -v '(none)' | wc -l)
+			}
+		install_command(){
+			package="$1"
+			run_command sudo apt-get install "$package"
+			}
+	elif [ "$os" = "Fedora" ]; then
+		validate_list="autoconf213 bison bzip2 ccache curl flex gawk gcc-c++ git glibc-devel glibc-static libstdc++-static libX11-devel make mesa-libGL-devel ncurses-devel patch zlib-devel ncurses-devel.i686 readline-devel.i686 zlib-devel.i686 libX11-devel.i686 mesa-libGL-devel.i686 glibc-devel.i686 libstdc++.i686 libXrandr.i686 zip perl-Digest-SHA wget"
+		validate_command(){
+			package="$1"
+			echo $(rpm -qa | egrep "$package" | wc -l)
+			}
+		install_command(){
+			package="$1"
+			run_command yum install "$package"
+			}
+	elif [ "$os" = "Arch Linux" ]; then
+		validate_list="alsa-lib autoconf2.13 bison ccache curl firefox flex gcc-multilib git gperf libnotify libxt libx11 mesa multilib-devel wget wireless_tools yasm zip lib32-mesa lib32-mesa-libgl lib32-ncurses lib32-readline lib32-zlib"
+		validate_command(){
+			package="$1"
+			echo $(pacman -Qs $a | wc -l)
+			}
+		install_command(){
+			package="$1"
+			run_command packman -S "$package"
+			}
+	else
+		echo "Cannot validate or install package, $os is not known"
+		exit 1
+	fi
+
+
+	# check to see wheher the packages are installed using the customized functions.
+	for a in $validate_list; do
+		installed=$(validate_command "$a")
+		if [ "$installed" -gt 1 ]; then
+			missing+="$a "
+		else
+			echo "Found $a already installed."
+		fi
+	done
+
+	# If there are any missing, install them.
+	echo $missing
+	if [ -z "$missing" ]; then
+		echo "All necessary packages installed"
+	else
+		echo "Installing missing packages: $missing"
+		echo
+		install_command "$missing"
+		echo
+	fi
+}
+
+
+bootstrap_linux() { 
+	if [ "$option_auto_install" = "yes" ]; then
+		auto_install
```

Comment:
When I run this command without any options, I get a prompt telling me that I'm ready to build B2G.  Given that there's nothing in this script other than installing packages, I wonder if we shouldn't just do this by default.  Also, you're checking which packages are already installed, so this seems like a harmless operation to do twice.

It would be cool if this bootstrapper had a thing to automatically add the udev rules for applicable USB ids including the Fastboot one.